slices/reduce.go

```diff
@@ -0,0 +1,25 @@
+package slices
+
+import "golang.org/x/exp/constraints"
+
+func ReduceWith[S ~[]E, E any, A any](s S, acc A, fn func(A, E) A) A {
+	for _, e := range s {
+		acc = fn(acc, e)
+	}
+
+	return acc
+}
+
+func Reduce[S ~[]E, E any, A any](s S, fn func(A, E) A) A {
```

Comment:
Honestly at this point, the heavy part of the API, to me, is the generic types. I think we can consider that the devs can pass the zero-value as accumulator themselves, don't you think? :slightly_smiling_face: